### PR TITLE
fix(monitoring): resolve circular import in chain generation

### PR DESCRIPTION
**Pull request by dewanshparashar**
## What broke                                                                                           
                                                                                                         
  `yarn workspace arb-token-bridge-ui generateOrbitChainsToMonitor` fails at module init with:            
                                                                                                          
```  
ReferenceError: Cannot access 'CommonUtils_1' before initialization                                     
    at getOrbitChains (src/util/orbitChainsList.ts:67)                                                    
    at TokenListUtils.ts:129                                                                              
 ```                                                                                                         
  This blocks monitoring-data generation. It started failing after #238.
                                                                                                          
  ## Why                                 
                                                                                                          
  #238 had a small incidental change in `TokenListUtils.ts`: a JSON import was swapped for a top-level    
  call to `getOrbitChains()`. That function lives on the other side of a pre-existing import cycle:
                                                                                                          
  TokenListUtils → orbitChainsList → AddressUtils → ... → TokenListUtils                                  
   
  During init, `getOrbitChains()` runs before `orbitChainsList` finishes importing `CommonUtils` — so     
  `isE2eTestingEnvironment` is still in the TDZ. Next.js's bundler hides this; ts-node doesn't.           
   
  ## Fix (one file, +2 / -19)                                                                             
                                         
  In `orbitChainsList.ts`:                                                                                
   
  - Convert `NativeCurrencyBase` and `ChainWithRpcUrl` to `import type` — they were always types, and     
  eliding them at runtime breaks the cycle.
  - Delete `getChainIdFromInboxAddress` (0 importers). This also drops the `AddressUtils` import, removing
   the other runtime path back into the cycle.                                                            
  - Delete `NetworkType` (0 importers).
                                                                                                          
  No other files changed.                                                                                 
   
  ## Regression guard                                                                                     
                                         
  Added `__tests__/orbitChainsList.test.ts` with 4 cases:

  - Calls `getOrbitChains()` — if a cycle reappears, the test file fails at collect time with the same    
  error the monitoring script would hit (verified by reverting the fix).
  - Mainnet-only and testnet-only filter correctness.                                                     
  - Every returned chain has the fields the monitoring script reads (`chainId`, `rpcUrl`,                 
  `ethBridge.inbox`).
                                                                                                          
  ## Verified                                                                                             
   
  - `yarn lint` clean                                                                                     
  - 4/4 tests pass - https://github.com/OffchainLabs/arbitrum-portal/actions/runs/24715988409/job/72292940341?pr=291#step:6:616                       
  - Monitoring script runs and produces output - https://github.com/OffchainLabs/arbitrum-portal/actions/runs/24715889556/job/72292078516

### CODE REVIEW

File: packages/arb-token-bridge-ui/src/util/orbitChainsList.ts [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
+import type { ChainWithRpcUrl } from './networks';
 import orbitChainsData from './orbitChainsData.json';
 
-export type NetworkType =
-  | 'Ethereum'
-  | 'Rollup'
-  | 'AnyTrust'
-  | 'Ethereum Testnet'
-  | 'Arbitrum Testnet';
-
```
Comment (dewanshparashar) [inline packages/arb-token-bridge-ui/src/util/orbitChainsList.ts:14]: Cleaned up dead exports

File: packages/arb-token-bridge-ui/src/util/orbitChainsList.ts [hunk 2]
```diff
@@ -76,14 +68,5 @@ export function getOrbitChains(
...
-  return (
-    getOrbitChains()
-      //
-      .find((chain) => chain.chainId === chainId)?.ethBridge.inbox
-  );
-}
-
-export function getChainIdFromInboxAddress(inboxAddress: string) {
-  return getOrbitChains().find((chain) => addressesEqual(chain.ethBridge.inbox, inboxAddress))
-    ?.chainId;
```
Comment (dewanshparashar) [inline packages/arb-token-bridge-ui/src/util/orbitChainsList.ts:88]: Dead exports cleaned and dependency (which was causing circular - `addressEqual`) was also removed.

File: packages/arb-token-bridge-ui/src/util/__tests__/orbitChainsList.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { describe, expect, it } from 'vitest';
+
+import { getOrbitChains, orbitMainnets, orbitTestnets } from '../orbitChainsList';
+
```
[the remark this answers is not in the document]
Comment (dewanshparashar) [reply packages/arb-token-bridge-ui/src/util/__tests__/orbitChainsList.test.ts:4]: Corner case + not urgent

**Review by fionnachan — APPROVED** [review]
